### PR TITLE
feat: add external secret store api and interface

### PR DESCRIPTION
**Pull request by adohe**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
add external secret store api and interface
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/secrets/types.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// ProviderSpec contains provider-specific configuration.
+type ProviderSpec struct {
+	// AWS configures a store to retrieve secrets from AWS Secrets Manager.
+	AWS *AWSProvider `yaml:"aws,omitempty" json:"aws,omitempty"`
+	// Vault configures a store to retrieve secrets from HashiCorp Vault.
+	Vault *VaultProvider `yaml:"vault,omitempty" json:"vault,omitempty"`
+}
+
+// AWSProvider configures a store to retrieve secrets from AWS Secrets Manager.
+type AWSProvider struct {
```
Comment (SparkYuan) [inline pkg/apis/secrets/types.go:17]: I am not sure whether the name `AWSProvider` is too general, considering this provider only supports one AWS product 'AWS Secrets Manager'

Comment (adohe) [reply pkg/apis/secrets/types.go:17]: > I am not sure whether the name `AWSProvider` is too general, considering this provider only supports one AWS product 'AWS Secrets Manager'

good question, but I think we can put the problem on the back for now, environment management also include aws related configurations, we can go back this after @healthjyk gives a detailed api reference.

Comment (SparkYuan) [reply pkg/apis/secrets/types.go:17]: okay

**Review by SparkYuan — APPROVED** [review]
lgtm

**Pull request merged**